lib/season_stats_module.rb

```diff
@@ -0,0 +1,170 @@
+module SeasonStatistics
+  require 'pry'
+
+  def find_all_games_from_season(season_id)
+    @games.find_all do |game|
+      game.season == season_id
+    end
+  end
+
+  def game_teams_data_for_season(season_id)
+    @game_teams.find_all do |game|
+      game.game_id[0..3] == season_id[0..3]
+    end
+  end
+
+  def season_coaches(season_id)
+    game_teams_data_for_season(season_id).map do |game|
+      game.head_coach
+    end.uniq
+  end
+
+  def winningest_coach(season_id)
+    coaches_hash = Hash.new
+    season_coaches(season_id).find_all do |coach|
+
+      total_games = game_teams_data_for_season(season_id).count do |game|
+        game.head_coach == coach
+      end
+
+      total_wins = game_teams_data_for_season(season_id).count do |game|
+        game.head_coach == coach && game.result == "WIN"
+      end
+      coaches_hash[coach] = ((total_wins.to_f/total_games.to_f) * 100).round(2)
+    end
+
+    best_coach = coaches_hash.max_by do |coach, win_percentage|
+      win_percentage
+    end[0]
+  end
+
+  def worst_coach(season_id)
+    coaches_hash = Hash.new
+    season_coaches(season_id).find_all do |coach|
+      total_games = game_teams_data_for_season(season_id).count do |game|
+        game.head_coach == coach
+      end
+
+      total_wins = game_teams_data_for_season(season_id).count do |game|
+        game.head_coach == coach && game.result == "WIN"
+      end
+      coaches_hash[coach] = ((total_wins.to_f/total_games.to_f) * 100).round(2)
+
+    end
+
+    worst_coach = coaches_hash.min_by do |coach, win_percentage|
+      win_percentage
+    end[0]
+  end
+
+  def most_accurate_team(season_id)
```

Comment:
Looks great! I like how you separated a lot of your data into helper methods.  Do we need the else clause of these if statements though? Maybe take a look at these for refactoring?